### DISCUSSION TITLE
Release 0.9

### DISCUSSION THREAD

**Discussion by sumeshir26** [sumeshir26/TimerX#27] [Announcements]
# Release 0.9

## What's Changed
* feat(including bugs): minor changes that increases functionality by @im-coder-lg in https://github.com/sumeshir26/TimerX/pull/1
* feat: Dark mode by @sumeshir26 in https://github.com/sumeshir26/TimerX/pull/2
* feat: Always on top by @sumeshir26 in https://github.com/sumeshir26/TimerX/pull/4
* fix: Actions by @sumeshir26 in https://github.com/sumeshir26/TimerX/pull/9
* Patch actions by @im-coder-lg in https://github.com/sumeshir26/TimerX/pull/12
* feat: Actions Improvements by @sumeshir26 in https://github.com/sumeshir26/TimerX/pull/13
* feat: Actions improvements(Part 2) by @sumeshir26 in https://github.com/sumeshir26/TimerX/pull/14
* feat: Unified action by @sumeshir26 in https://github.com/sumeshir26/TimerX/pull/19
* feat: New Icon +Sun Valley theming in configurator.py by @not-nef in https://github.com/sumeshir26/TimerX/pull/18
* patch(readme): readme spelling errors and more by @im-coder-lg in https://github.com/sumeshir26/TimerX/pull/21
* feat(surprise): Website by Docusaurus by @im-coder-lg in https://github.com/sumeshir26/TimerX/pull/26
* Settings Window by @not-nef in #25 

## New Contributors
* @im-coder-lg made their first contribution in https://github.com/sumeshir26/TimerX/pull/1
* @sumeshir26 made their first contribution in https://github.com/sumeshir26/TimerX/pull/2
* @not-nef made their first contribution in https://github.com/sumeshir26/TimerX/pull/18

**Full Changelog**: https://github.com/sumeshir26/TimerX/compare/v0.1...0.9

<hr /><em>This discussion was created from the release <a href='https://github.com/sumeshir26/TimerX/releases/tag/0.9'>Release 0.9</a>.</em>

**Comment by lg-nineoneninethreenine** [comment]
Python Tk+noVNC fails, looped you in the issue.

**Reply by sumeshir26** [reply]
Yeah, got the notiification...

**Reply by lg-nineoneninethreenine** [reply]
I kinda got the fix. At https://github.com/gitpod-io/workspace-images/pull/303(go to forum link in first comment) they told that you need to install Tk before `pyenv` which manages Python versions. I found [this](https://www.tecmint.com/pyenv-install-and-manage-multiple-python-versions-in-linux/) guide and am using/testing that in another branch. Lots of lines(10-11 is too much) in a shell script(not a whole lot though).

**Reply by lg-nineoneninethreenine** [reply]
I started my work _here_, apologies for adding 300 lines of Dockerfile, no choice that that. I need to copy more files and we could be good to go.

**Comment by not-nef** [comment]
my pr isnt featured :(

**Reply by lg-nineoneninethreenine** [reply]
Adding that!

**Reply by lg-nineoneninethreenine** [reply]
How's that as a matter of fact?

**Reply by lg-nineoneninethreenine** [reply]
Added on the release too :)

**Reply by sumeshir26** [reply]
@not-nef Hi, your pr is not included in the 0.9 release, it will be included in the next release afterwards I make some minor changes.

**Reply by not-nef** [reply]
epic

**Reply by lg-nineoneninethreenine** [reply]
Have you clarified it in the release(changed it)?

**Reply by not-nef** [reply]
he said it isnt featured in the 0.9 release, so it isnt listed in the changelog

**Reply by lg-nineoneninethreenine** [reply]
Well, I changed it so there's supposed to be a change right?

**Comment by lg-nineoneninethreenine** [comment]
I think this is supposed to be pinned on the Discussions tab. What do you think?

**Comment by not-nef** [comment]
Can someone remove the macos build from the release pls it doesnt even work

**Reply by sumeshir26** [reply]
I mean it doesn't hurt to keep it... Also, it might work now due to a cx_freeze update

**Reply by not-nef** [reply]
I thought we all agreed to dropping macos?

**Reply by sumeshir26** [reply]
Yeah macOS is dropped, but it doesent hurt if we give the DMG, It doesn't mean any extra work for us.

**Reply by not-nef** [reply]
But it doesnt work!!

**Reply by not-nef** [reply]
Why should we give non-working builds?

**Reply by sumeshir26** [reply]
It should theoretically work on big sur now now that GH actions works on macOS 11. There is a bug with cx_freeze that makes the DMG work _only_ on the macOS version it was compiled on. Now that GH Actions uses Big Sur, it should work. Back when you tested it on your dad's mac, GH actions used Catalina

**Reply by lg-nineoneninethreenine** [reply]
Would it work now?
I think we should add the `Upload Asset` code to the normal workflows and have a deprecated macOS workflow so that we could add(and eventually test) the Mac builds.

**Reply by lg-nineoneninethreenine** [reply]
Also, we need to use `macos-11` to use Big Sur.

**Reply by sumeshir26** [reply]
@im-coder-lg Recently(Since ~3 Weeks) `macos-latest` uses `macos-11`

**Reply by lg-nineoneninethreenine** [reply]
Okay :headpalm:

**Reply by lg-nineoneninethreenine** [reply]
I added Big Sur with `macos-11` and Catalina with `macos-10.15`. If GH supports High Sierra, we could try testing with that too!
